`create_tag` cannot create `<header>` element

When I attempt to create a `<header>` element, a `<head>` element is created instead:

```pycon
>>> from selectolax.parser import create_tag
>>> header = create_tag("header")
>>> header
<Node head>
```

Example test function in `test_utils.py`:

```python
@pytest.mark.parametrize(*_IMPL_PARAMETRIZER)
def test_create_header_tag(impl: Impl):
    node = impl.tag_fn("header")
    assert isinstance(node, impl.node)
    assert node.html == "<header></header>"
```

Test output:

```
FAILED tests/test_utils.py::test_create_header_tag[impl0] - AssertionError: assert '<head></head>' == '<header></header>'
FAILED tests/test_utils.py::test_create_header_tag[impl1] - AssertionError: assert '<head></head>' == '<header></header>'
```

--- a/selectolax/utils.pxi
+++ b/selectolax/utils.pxi
@@ -34,7 +34,7 @@ def get_fragment_type(
         tree = parser_cls(html)
     
     import re
-    html_re = re.compile(r"<html|<body|<head", re.IGNORECASE)
+    html_re = re.compile(r"<html|<body|<head(?!er)", re.IGNORECASE)
 
     has_html = False
     has_head = False

--- a/tests/test_utils.py
+++ b/tests/test_utils.py
@@ -49,6 +49,12 @@ def test_create_tag(impl: Impl):
     assert node.html == "<p></p>"
 
 
+@pytest.mark.parametrize(*_IMPL_PARAMETRIZER)
+def test_create_header_tag(impl: Impl):
+    node = impl.tag_fn("header")
+    assert isinstance(node, impl.node)
+    assert node.html == "<header></header>"
+
 # Cases to test parse_fragment():
 # - <doctyle> + <html> only
 # - HTML with <head>

